crafters/image/ImageTorchTransformation/__init__.py

```diff
@@ -0,0 +1,168 @@
+            crafted_imgs = crafter.craft(image_batch)
+
+        The torch equivalent for this would be recursively apply `Compose` on each image.
+
+        .. code-block:: python
+            import torchvision.transforms as T
+
+            transforms = T.Compose(
+                [
+                    T.ToTensor(),
+                    T.CenterCrop(300),
+                    T.Resize((244, 244)),
+                    T.RandomVerticalFlip(p=1.0),
+                    T.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
+                ]
+            )
+
+            crafted_img = transforms(image_batch[0]).cpu().numpy()
+            crafted_img = np.transpose(crafted_img, (1, 2, 0))
+
+
+    :param transforms: List of transforms that are applied sequentially
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        transforms: List[Union[str, Dict[str, Dict]]] = [
+            {'Resize': dict(size=(224, 224))},
+            {'Normalize': dict(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225))},
+        ],
+        *args,
+        **kwargs,
+    ):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        if not isinstance(transforms, list):
+            self.logger.error('The `transforms` argument has to be a list.')
+            raise ValueError
+
+        self.transforms_specification = transforms
+
+    def post_init(self):
+        from torchvision import transforms as T
+
+        transforms_list = []
+        for transform in self.transforms_specification:
+            if isinstance(transform, str):
+                tr_name, tr_kwargs = transform, {}
+            elif isinstance(transform, dict):
+                tr_name, tr_kwargs = next(iter(transform.items()))
+            else:
+                self.logger.error(
+                    'Each element of the `transform` has to be either a dict or  a string'
+                )
+                raise ValueError
+
+            try:
+                tr_class = getattr(T, tr_name)
+            except AttributeError:
+                self.logger.error(
+                    f'The transform class `{tr_name}` does not seem to exist,'
+                    'check that you have not made a typo.'
+                )
+                raise ValueError
+
+            try:
+                transform_instance = tr_class(**tr_kwargs)
+            except:
+                self.logger.error(
+                    f'Error instantiating torchvision transforms class `{tr_name}`,'
+                    ' check if all kwargs are valid and if transform is valid for tensors.'
+                )
+                raise ValueError
+
+            # Removes randomness in transforms
+            # does not handle inherently random transform like `RandomErasing`
+            # where p is probabilty of being random
+            transform_instance.__dict__.update({'p': 1.0})
```

Comment:
> The stage at which I made `p`=1 was before initializing the transform and having an invalid kwarg throws error, tried it and is also testcase. Instead what I did now is to inject into the transform variables with dictionary `update`. This worked and looks neat, I guess. If `p` is used its 1 now, or its just an unwanted value it holds.

That is what I meant here. So if you want it in `tr_kwargs` then I have check if the class actually takes `p` as argument with the not so pretty code -`if "p" in tr_class.__init__.__code__.co_varnames:`

Reply:
why is it necessary? normally the inits accept extra kwargs that are just ignored, have u even tried?

Reply:
It is necessary because by default most transforms have `p` set to 0.5 by default which means it may not always be applied. We make sure it is deterministic (across runs) by setting it to 1.

Yes, I sure tried again after you said to yesterday. That is also why the initialization is under try-catch (even in albumentations) and have also included test cases for wrong kwargs as well. 

Below are couple of screenshots showing it 

![image](https://user-images.githubusercontent.com/17333185/116448667-a93f8480-a859-11eb-9d93-d2e19898c8ef.png)
![image](https://user-images.githubusercontent.com/17333185/116449558-95e0e900-a85a-11eb-8781-e4fefd58a571.png)

Reply:
ok, thanks for the explanation

Reply:
I understand your pov. It has something to do with the transform class implementation not having **kwargs as a parameter.